captum/attr/_core/noise_tunnel.py

```diff
@@ -79,13 +80,16 @@ def multiplies_by_inputs(self):
     @noise_tunnel_n_samples_deprecation_decorator
     def attribute(
         self,
-        inputs: Union[Tensor, Tuple[Tensor, ...]],
+        inputs: TensorOrTupleOfTensorsGeneric,
```

Comment:
That's a good point! I assumed that, that is the reason why we don't have overrides for different attribute signatures in this case. 
This method was also lacking return type hint which I added also using `TensorOrTupleOfTensorsGeneric`. I was thinking that we can use `TensorOrTupleOfTensorsGeneric` shortcut instead of `Union[Tensor, Tuple[Tensor, ...]]`. But it looks like there needs to be an agreement between inputs and output because `TensorOrTupleOfTensorsGeneric` is generics. I that case I should probably also avoid using it for the output type.